src/main/scala/gitbucket/core/controller/api/ApiIssueMilestoneControllerBase.scala

```diff
@@ -0,0 +1,82 @@
+package gitbucket.core.controller.api
+import gitbucket.core.api._
+import gitbucket.core.controller.ControllerBase
+import gitbucket.core.service.{AccountService, IssueCreationService, IssuesService, MilestonesService}
+import gitbucket.core.util.{ReadableUsersAuthenticator, ReferrerAuthenticator}
+import gitbucket.core.util.Implicits._
+
+trait ApiIssueMilestoneControllerBase extends ControllerBase {
+  self: AccountService
+    with IssuesService
+    with IssueCreationService
+    with MilestonesService
+    with ReadableUsersAuthenticator
+    with ReferrerAuthenticator =>
+
+  /*
+   * i. List milestones
+   * https://docs.github.com/en/rest/reference/issues#list-milestones
+   */
+  get("/api/v3/repos/:owner/:repository/milestones")(referrersOnly { repository =>
+    val state = params.getOrElse("state", "all")
+    // TODO "sort", "direction" params should be implemented.
+    val apiMilestones = (for ((milestoneWithIssue, i) <- getMilestonesWithIssueCount(repository.owner, repository.name)
+                                .sortBy(p => p._1.milestoneId)
+                                .zipWithIndex)
+      yield {
+        ApiMilestone(
+          repository.repository,
+          milestoneWithIssue._1,
+          ApiUser(context.loginAccount.get),
+          i + 1,
+          milestoneWithIssue._2,
+          milestoneWithIssue._3
+        )
+      }).reverse
+    state match {
```

Comment:
As there is no index correspond to milestone `number` of each repository, I use zipIndex to put number after sorting by `milestoneId` in repository. But actually if some milestones deleted, `number` will be different...? So means this way is not immutable...
Maybe number column is required?? or use `milestoneId` as `number`??

Reply:
I'm not sure the exact meaming of `number` in GitHub API. However, we use `issueId` as `number` in Issue/Pull request APIs. Therefore, using `milestoneId` as `number` makes sense.
https://github.com/gitbucket/gitbucket/blob/f6a5def6386da6d82ffec3af25cdc7366fd456f5/src/main/scala/gitbucket/core/api/ApiIssue.scala#L48-L49

But why `id = 0` in Issue/Pull request APIs...? 🤔 
https://github.com/gitbucket/gitbucket/blob/f6a5def6386da6d82ffec3af25cdc7366fd456f5/src/main/scala/gitbucket/core/api/ApiIssue.scala#L22

Reply:
@takezoe Ok! So I will fix PR to use `milestoneId` as `number`.
About `id = 0`, I don't know neither. It is not in use. so we probably can remove...

Reply:
Yeah... Regarding Issue/Pull request API, I think `id` field is necessary, but we need to look into the reason for the dummy value, this is another topic though.

Reply:
@takezoe I fixed PR (added delete endpoint)

Reply:
Thanks!